src/profile-logic/perf-script-profile-format.js

```diff
@@ -0,0 +1,219 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+/**
+ * The "perf script" format is the plain text format that is output by an
+ * invocation of `perf script`, where `perf` is the Linux perf command line tool.
+ */
+export function isPerfScriptFormat(profile: string): boolean {
+  const firstLine = profile.substring(0, profile.indexOf('\n'));
+  return /^(\S.+?)\s+(\d+)\/*(\d+)*\s+([\d.]+)/.test(firstLine);
+}
+
+/**
+ * Convert the output from `perf script` into the gecko profile format.
+ */
+export function convertPerfScriptProfile(profile: string): Object {
```

Comment:
I think this would be more maintainable if we targeted the processed profile format. Our types are more robust for that format (the schema objects are difficult to make type safe), and it is closer to the format we actually want it in. We can get really nice type errors for format changes if we provide a type for each object created:

```
const markersTable: MarkersTable = {

};
```

And also if we return a `Profile` type from the function it will ensure that this code will be easy to maintain.

Reply:
It also means that we'll *need* to maintain it every time we change the profile format. Maybe that's ok, though. But this implementation has the advantage of not needing to be maintained because it'll just go through profile upgrading.

Reply:
That's true... Hmm... I guess I'm not terribly opinionated then, although I think we should explicitly set a gecko profile version if we go that route. I'd be fine with either way you want to do it @jesup.